marlowe-cli/src/Language/Marlowe/CLI/Transaction.hs

```diff
@@ -752,7 +752,8 @@ publishImpl connection signingKey expires changeAddress publishingStrategy coinS
     coinSelectionStrategy
     printStats
 
-  void $ submitBody connection txBody [signingKey] timeout
```

Comment:
I would keep here the previous semantics - the function is called `publishImpl` and not `build*Impl` - it's main purpose is to actually do the publishing ;-)

We should just skip a call on our side when we don't have a `timeout` at hand.